opensearch-project/OpenSearch-Dashboards
# Fix HeaderControl not rendered if not mount in initial rendering

### Description


This PR is for fixing `HeaderControl` not rendered if not mount in initial rendering. The original implementation rely on controls registered to `application.currentRightControls$` before `Header` component rendering. This component won't re-render after new value emitted in `application.currentRightControls$`. This PR add a new `useObservableValue` to subscribe these controls updates. It will fire re-render after these observable emit new values.

### Issues Resolved



#8313 

## Screenshot


This is a bug fix, no UI changes.

## Testing the changes



- Clone branch code and run `yarn osd bootstrap --single-version loose`
- Add below configs in `config/opensearch_dashboards.yml`
```
savedObjects.permission.enabled: true
workspace.enabled: true
uiSettings:
  overrides:
    'home:useNewHomePage': true
opensearchDashboards.dashboardAdmin.users: ['admin']
```
- Run `yarn start --no-base-path`
- Visiting a existing workspace settings page with admin user
- The delete button should be displayed

## Changelog

- fix: Fix HeaderControl not rendered if not mount in initial rendering

### Check List
